app/src/main/java/ch/epfl/sweng/swenggolf/database/FakeDatabase.java

```diff
@@ -185,6 +186,29 @@ public void readOffers(@NonNull ValueListener<List<Offer>> listener,
                 listener, categories);
     }
 
+    @Override
+    public void readFollowers(@NonNull ValueListener<Map<String, List<String>>> listener) {
+        Map<String, List<String>> userFollowing = new HashMap<>();
+        for (Map.Entry<String, Object> entry : database.entrySet()) {
+            if (entry.getKey().startsWith(FOLLOWERS_PATH)) {
+                getFollowersRepertory(userFollowing, entry);
+            }
+        }
+        FakeDatabaseListHandler.readFollowers(working, listener, userFollowing);
+    }
+
+    private void getFollowersRepertory(Map<String, List<String>> userFollowing,
+                                       Map.Entry<String, Object> entry) {
+        String s = entry.getKey().substring(1); // to remove first '/' char
```

Comment:
Isn't it better to do the split and then to just add to all the magic numbers that follow ? Maybe you might add some final values in the method to clarify the code.

Reply:
Well, I know that there are some magic numbers, and that overall it's not really clean, but I had no other choice since we just have to make it work for the FakeDatabase. I don't understand what you mean by "split and then add to all the magic numbers" though. I am not convinced that declaring constants would be any better but if you prefer it, I'll change it.

Reply:
I mean that you instead of getting the sub-string just don't take into account your element 0 in the array that is returned by split. And your 2 becomes a 3 and your children[1] becomes a children[2]. When you're testing the length you test it to avoid getting elements that have sub-elements ?